From 9c6ac58e4fcb4b3fd2cda943f4680ff5a5d80f82 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 18 May 2020 16:57:25 -0700
Subject: [PATCH] hciuart: lower the speed and restart on failure

:Release Notes:
hciuart: lower the speed and restart on failure

:Detailed Notes:
The baud rate is set to 460800 bits per second
Restart service on failure

:Testing Performed:
Only build tested.

:QA Notes:
N/A

:Issues Addressed:
[WRQ-10121]
---
Upstream-Status: Inappropriate [webOS specific]

 debian/pi-bluetooth.hciuart.service | 1 +
 usr/bin/btuart                      | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/debian/pi-bluetooth.hciuart.service b/debian/pi-bluetooth.hciuart.service
index 5c866de..b9ee6a5 100644
--- a/debian/pi-bluetooth.hciuart.service
+++ b/debian/pi-bluetooth.hciuart.service
@@ -6,6 +6,7 @@ After=dev-serial1.device
 [Service]
 Type=forking
 ExecStart=/usr/bin/btuart
+Restart=on-failure
 
 [Install]
 WantedBy=dev-serial1.device
diff --git a/usr/bin/btuart b/usr/bin/btuart
index 493fd86..cb99a4f 100755
--- a/usr/bin/btuart
+++ b/usr/bin/btuart
@@ -25,7 +25,7 @@ if [ "$uart0" = "$serial1" ] ; then
 	if [ "$uart0_pins" = "16" ] ; then
 		$HCIATTACH /dev/serial1 bcm43xx 3000000 flow - $BDADDR
 	else
-		$HCIATTACH /dev/serial1 bcm43xx 921600 noflow - $BDADDR
+		$HCIATTACH /dev/serial1 bcm43xx 460800 noflow - $BDADDR
 	fi
 else
 	$HCIATTACH /dev/serial1 bcm43xx 460800 noflow - $BDADDR
